### PR TITLE
POC for specifying the Prisma Cloud API URL as a command-line option

### PR DESCRIPTION
**Pull request by tkishel**
UNTESTED POC

Ensuring CLI ( `--prisma_api_url`) / ENV ('PRISMA_API_URL') equivalency.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -141,7 +139,8 @@ def setup_http_manager(self, ca_certificate=os.getenv('BC_CA_BUNDLE', None)):
                 self.http = urllib3.PoolManager()
 
     def setup_bridgecrew_credentials(self, repo_id, skip_fixes=False, skip_suppressions=False,
-                                     skip_policy_download=False, source=None, source_version=None, repo_branch=None):
+                                     skip_policy_download=False, source=None, source_version=None, repo_branch=None,
+                                     prisma_api_url=os.getenv('PRISMA_API_URL', None)):
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:143]: I don't think we need the default value here, because it will just get called from main.py using the config value.

File: checkov/main.py [hunk 2]
```diff
@@ -401,6 +402,8 @@ def add_parser_args(parser):
     parser.add('--skip-cve-package',
                help='filter scan to run on all packages but a specific package identifier (denylist), You can '
                     'specify this argument multiple times to skip multiple packages', action='append', default=None)
+    parser.add('--prisma_api_url',
```
Comment (mikeurbanski1) [inline checkov/main.py:405]: I think this should go up next to the API key option.

File: checkov/common/bridgecrew/platform_integration.py (hunk 1 above)
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:143]:
```suggestion
                                     prisma_api_url):
```

I don't think we need the default value here, because it will just get called from main.py using the config value.

File: checkov/main.py (hunk 2 above)
Comment (gruebel) [inline checkov/main.py:405]:
```suggestion
    parser.add('--prisma-api-url',
```
to keep it in line with the other parameters 🙂

Comment (tkishel) [reply checkov/main.py:405]: Done.

File: checkov/common/bridgecrew/platform_integration.py (hunk 1 above)
Comment (tkishel) [reply checkov/common/bridgecrew/platform_integration.py:143]: Done.

File: checkov/main.py (hunk 2 above)
Comment (tkishel) [reply checkov/main.py:405]: Done.

**Comment by tkishel** [comment]
I've also normalized the associated help and error messages.

**Comment by nimrodkor** [comment]
@tkishel To support backwards compatibility, can we still fall back to the env variable if the arg was not set?

**Comment by tkishel** [comment]
> To support backwards compatibility, can we still fall back to the env variable if the arg was not set?

Yes, via `parser.add` specifying `env_var` here: 

https://github.com/bridgecrewio/checkov/pull/2297/files#diff-83a7665ffac295dc2773f5e32e4c73d78da1189fc93f8365b69b61a0a17d6c50R344

Resulting in:

```
tkishel # PRISMA_API_URL=https://api.prismacloud.io checkov -f example/main.tf --bc-api-key xxx
<snip>
terraform scan results:

Passed checks: 9, Failed checks: 36, Skipped checks: 0
```

And:

```
tkishel # checkov -f example/main.tf --prisma-api-url https://api.prismacloud.io --bc-api-key xxx
<snip>
terraform scan results:

Passed checks: 9, Failed checks: 36, Skipped checks: 0
```

**Comment by tkishel** [comment]
IDK why the `CodeQL / Analyze (python) (pull_request)` check is failing, but its configuration differs from another recent pull request in that a `CODEQL_PYTHON: /root/.local/share/virtualenvs/checkov-m3Ue6TPK/bin/python` environment variable is being set here:

```
CodeQL / Analyze (python) (pull_request)

Perform CodeQL Analysis

Run github/codeql-action/analyze@v1
  with:
    output: ../results
    upload: true
    cleanup-level: brutal
    add-snippets: false
    skip-queries: false
    checkout_path: /actions-runner/_work/checkov/checkov
    upload-database: true
    wait-for-processing: false
    token: ***
    matrix: {
    "language": "python"
  }
  env:
    CODEQL_ACTION_RUN_MODE: Action
    CODEQL_ACTION_VERSION: 1.0.30
    CODEQL_ACTION_FEATURE_SARIF_COMBINE: true
    CODEQL_ACTION_FEATURE_WILL_UPLOAD: true
    CODEQL_ACTION_ANALYSIS_KEY: .github/workflows/codeql-analysis.yml:analyze
    CODEQL_WORKFLOW_STARTED_AT: 2022-01-30T06:47:24.063Z
    CODEQL_ACTION_FEATURE_MULTI_LANGUAGE: false
    CODEQL_ACTION_FEATURE_SANDWICH: false
    CODEQL_PYTHON: /root/.local/share/virtualenvs/checkov-m3Ue6TPK/bin/python
    CODEQL_RAM: 6831
    CODEQL_THREADS: 2
```

Resulting in an error unrelated to the changes in the commits of this pull request:

```
Extracting python
  /root/.local/share/virtualenvs/checkov-m3Ue6TPK/bin/python -c import os; import pip; print(os.path.dirname(os.path.dirname(pip.__file__)))
  /root/.local/share/virtualenvs/checkov-m3Ue6TPK/bin/python: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
  Error: The process '/root/.local/share/virtualenvs/checkov-m3Ue6TPK/bin/python' failed with exit code 127
```

**Comment by tkishel** [comment]
Updated the unit tests, and normalized the associated instance variable (and renamed the associated `normalize_prisma_url` function that normalizes that variable, as it is not specific to a Prisma Cloud URL).

**Comment by nimrodkor** [comment]
Hey @tkishel !

It seems we still have a UT that is failing:
```
test_overriding_pc_api_url
```

Can you take a look at it?

**Comment by tkishel** [comment]
Ah, sorry. Should be fixed with the most recent commit.

**Comment by tkishel** [comment]
> Updated the unit tests, and normalized the associated instance variable (and renamed the associated `normalize_prisma_url` function that normalizes that variable, as it is not specific to a Prisma Cloud URL).

FYI: I added logic from this code, which represents unfortunately common user errors when specifying Prisma Cloud API URLs:

https://github.com/PaloAltoNetworks/pcs-toolbox/blob/main/pc_lib/pc_lib_utility.py#L144

That should prevent some simple support cases.

**Comment by tkishel** [comment]
IDK about the bandit action, but ...

```
# tkishel # Code # checkov # git branch
  master
* poc_prisma_api_url_cli

# tkishel # Code # checkov # git log
commit 88292ad45bb85968dee39af361b2963867bf924d (HEAD -> poc_prisma_api_url_cli, origin/poc_prisma_api_url_cli)
Author: tkishel <tkishel@gmail.com>
Date:   Wed Feb 16 10:06:08 2022 -0800

    fix testing after adding passing prisma_api_url to setup_bridgecrew_credentials

# tkishel # Code # checkov # bandit -r checkov
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	running on Python 3.10.2
1122 [0.. 50.. 100.. 150.. 200.. 250.. 300.. 350.. 400.. 450.. 500.. 550.. 600.. 650.. 700.. 750.. 800.. 850.. 900.. 950.. 1000.. 1050.. 1100.. ]
Run started:2022-02-17 15:48:26.059249

Test results:
	No issues identified.

Code scanned:
	Total lines of code: 32868
	Total lines skipped (#nosec): 53

Run metrics:
	Total issues (by severity):
		Undefined: 0.0
		Low: 0.0
		Medium: 0.0
		High: 0.0
	Total issues (by confidence):
		Undefined: 0.0
		Low: 0.0
		Medium: 0.0
		High: 0.0
Files skipped (0):

tkishel # Code # checkov # git log
commit 88292ad45bb85968d
```
